File: backend/src/models/Category.js
const mongoose = require('mongoose');

const categorySchema = new mongoose.Schema(
  {
    title: { type: String, required: true },
    slug: { type: String, required: true, unique: true },
    description: { type: String, default: '' },
    imageUrl: { type: String, default: '' },
    order: { type: Number, default: 0 },
    highlightedText: { type: String, default: '' },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Category', categorySchema);

File: frontend/lib/models/category.dart
class Category {
  const Category({
    required this.id,
    required this.title,
    required this.slug,
    this.description,
    this.imageUrl,
    this.highlightedText,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['_id'] as String,
      title: json['title'] as String? ?? '',
      slug: json['slug'] as String? ?? '',
      description: json['description'] as String?,
      imageUrl: json['imageUrl'] as String?,
      highlightedText: json['highlightedText'] as String?,
    );
  }

  final String id;
  final String title;
  final String slug;
  final String? description;
  final String? imageUrl;
  final String? highlightedText;
}

